common/src/main/java/org/opensearch/ml/common/MLPrompt.java

```
@@ -0,0 +1,202 @@
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@EqualsAndHashCode
+public class MLPrompt implements ToXContentObject, Writeable {
+
+    public static final String PROMPT_ID_FIELD = "prompt_id";
+    public static final String NAME_FIELD = "name";
+    public static final String DESCRIPTION_FIELD = "description";
+    public static final String PROMPT_FIELD = "prompt";
+    public static final String TAG_FIELD = "tag";
+    public static final String CREATE_TIME_FIELD = "create_time";
+    public static final String LAST_UPDATE_TIME_FIELD = "last_update_time";
+
+    @Setter
+    private String promptId;
+    private String name;
+    private String description;
+    @Setter
+    private Map<String, String> prompt;
+    private String tag;
+    private String tenantId;
+    @Setter
+    private Instant createTime;
+    @Setter
+    private Instant lastUpdateTime;
+
+    @Builder(toBuilder = true)
+    public MLPrompt(
+        String promptId,
+        String name,
+        String description,
+        Map<String, String> prompt,
+        String tag,
+        String tenantId,
+        Instant createTime,
+        Instant lastUpdateTime
+    ) {
+        this.promptId = promptId;
+        this.name = name;
+        this.description = description;
+        this.prompt = prompt;
+        this.tag = tag;
+        this.tenantId = tenantId;
+        this.createTime = createTime;
+        this.lastUpdateTime = lastUpdateTime;
+    }
+
+    public MLPrompt(StreamInput input) throws IOException {
+        Version streamInputVersion = input.getVersion();
```

is the version a mandatory field?

---

Yes, I think the version should be a mandatory field, and should throw an error if the version field is not provided by the user.

---

can we do version has a default one if no version is required? I would assume when I don't put a version, it has an initial version.

---

Okay, I will make the version field optional. Thank you!